### PR TITLE
Rust 1.45.0 release announcement

### PR DESCRIPTION
**Pull request by steveklabnik**
cc @rust-lang/release @rust-lang/core 

Also, cc @SergioBenitez since Rocket features here. Please let me know if you'd like anything changed!

### CODE REVIEW

File: posts/2020-07-16-Rust-1.45.0.md [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+```rust
+let x: f32 = 1.0;
+let y: u8 = unsafe { x.to_int_unchecked() };
+```
+
+But as always, you should only use this method as a last resort. Just like
+with array access, the compiler can often optimize the checks away, making
+the safe and unsafe versions equivalent when the compiler can prove it.
+
+## Stabilizing function-like procedrual macros in expressions, patterns, and statements
```
Comment (Manishearth) [inline posts/2020-07-16-Rust-1.45.0.md:150]:
```suggestion
## Stabilizing function-like procedural macros in expressions, patterns, and statements
```

File: posts/2020-07-16-Rust-1.45.0.md [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+fn hello(name: String, age: u8) -> String {
+    format!("Hello, {} year old named {}!", age, name)
+}
+
+fn main() {
+    rocket::ignite().mount("/", routes![hello]).launch();
+}
+```
+
+At the same time, Rocket was only able to achieve its power and ergonomics by
```
Comment (Manishearth) [inline posts/2020-07-16-Rust-1.45.0.md:223]: the "at the same time" feels weird here

File: posts/2020-07-16-Rust-1.45.0.md [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+
+In the end, the decision was made to do this:
+
+* `as` would perform a "saturating cast".
+* A new `unsafe` cast would be added if you wanted to skip the checks.
+
+This is very similar to array access, for example:
+
+* `array[i]` will check to make sure that `array` as at least `i` elements
+* You can use `unsafe { array.get_unchecked(i) }` to skip the check
```
Comment (XAMPPRocky) [inline posts/2020-07-16-Rust-1.45.0.md:107]:
```suggestion
* `array[i]` will check to make sure that `array` has at least `i` elements.
* You can use `unsafe { array.get_unchecked(i) }` to skip the check.
```

File: posts/2020-07-16-Rust-1.45.0.md (hunk 2 above)
Comment (XAMPPRocky) [reply posts/2020-07-16-Rust-1.45.0.md:223]: I agree, I think the sentence works just as well without it.
```suggestion
Rocket was only able to achieve its power and ergonomics by
```

File: posts/2020-07-16-Rust-1.45.0.md [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+```
+
+But as always, you should only use this method as a last resort. Just like
+with array access, the compiler can often optimize the checks away, making
+the safe and unsafe versions equivalent when the compiler can prove it.
+
+## Stabilizing function-like procedrual macros in expressions, patterns, and statements
+
+In [Rust 1.30.0](https://blog.rust-lang.org/2018/10/25/Rust-1.30.0.html), we stabilized
+"function-like procedrual macros in item position." For example, [the
```
Comment (CryZe) [inline posts/2020-07-16-Rust-1.45.0.md:153]:
```suggestion
"function-like procedural macros in item position." For example, [the
```

File: posts/2020-07-16-Rust-1.45.0.md [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+
+Being able to use macros in more places is interesting, but there's another
+reason why many Rustaceans have been waiting for this feature for a long
+time. [Rocket](https://rocket.rs/) is a popular web framework for Rust.
+Initially released in December of 2016, it has often been described as one of
+the best things the Rust ecosystem has to offer. Here's the "hello world"
+example from its homepage:
+
+```rust
+#![feature(proc_macro_hygiene, decl_macro)]
```
Comment (SergioBenitez) [inline posts/2020-07-16-Rust-1.45.0.md:209]: Would it be better to use the expected example from `0.5`?

```rust
#[macro_use] extern crate rocket;

#[get("/<name>/<age>")]
fn hello(name: String, age: u8) -> String {
    format!("Hello, {} year old named {}!", age, name)
}

#[launch]
fn rocket() -> rocket::Rocket {
    rocket::ignite().mount("/hello", routes![hello])
}
```

If so, perhaps the text above could change to:

> Being able to use macros in more places is interesting, but there's another
reason why many Rustaceans have been waiting for this feature for a long time:
[Rocket](https://rocket.rs/). Inititally released in December of 2016, Rocket is
a popular web framework for Rust often described as one of the best things the
Rust ecosystem has to offer. Here's the "hello world" example from its upcoming release:

Comment (pietroalbini) [reply posts/2020-07-16-Rust-1.45.0.md]: @SergioBenitez done!

Comment (SergioBenitez) [reply posts/2020-07-16-Rust-1.45.0.md]: The next paragraph will also need to change to accommodate this change. Drafting a suggestion now.

File: posts/2020-07-16-Rust-1.45.0.md [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+
+Rocket was only able to achieve its power and ergonomics by
+using nightly-only features. You can see two of them in the example above.
+However, as you may guess from the name of those two features, these are the
+ones that are finally stable! [This
+issue](https://github.com/SergioBenitez/Rocket/issues/19) has been tracking
+the history of all of the various features, and now they're all checked off!
+
+Rocket 0.5 has not yet been released, but when it is, many folks will be very
+happy :)
```
Comment (SergioBenitez) [inline posts/2020-07-16-Rust-1.45.0.md:230]:
```suggestion
Until today, Rocket depended on nightly-only features to deliver on its promise
of flexibility and ergonomics. In fact, as can be seen on the [project's
homepage](https://rocket.rs/v0.4), the same example above in the current version
of Rocket requires the `proc_macro_hygiene` feature to compile. However, as you
may guess from the feature's name, today it ships in stable! [This
issue](https://github.com/SergioBenitez/Rocket/issues/19) tracked the history of
nightly-only features in Rocket. Now, they're all checked off!

This next version of Rocket is still in the works, but when released, many folks
will be very happy :)
```

**Comment by steveklabnik** [comment]
🎊

**Pull request merged**